### PR TITLE
KeystoneAPI: print columns for Status/Reason

### PR DESCRIPTION
**Pull request by dprince**

### CODE REVIEW

File: api/v1beta1/keystoneapi_types.go [hunk 1]
```diff
@@ -21,6 +21,24 @@ import (
...
+	KeystoneAPITypeDbSync condition.Type = "DbSync"
+
+	// condition reasons
+	//
+
+	// KeystoneAPIReasonProvisioned - keystoneAPI provisioned
+	KeystoneAPIReasonProvisioned condition.Reason = "KeystoneAPIProvisioned"
+
+	// KeystoneAPIReasonDatabaseSync - keystoneAPI database sync
+	KeystoneAPIReasonDatabaseSync condition.Reason = "KeystoneAPIDatabaseSync"
```
[the remark this answers is not in the document]
Comment (stuggi) [reply api/v1beta1/keystoneapi_types.go:39]: or probably better add the DB related ones to https://github.com/openstack-k8s-operators/lib-common/blob/master/pkg/database/conditions.go

[the remark this answers is not in the document]
Comment (dprince) [reply api/v1beta1/keystoneapi_types.go:39]: @stuggi re making them generic in lib-common I was actually thinking of making more of the status codes there reside in each operator (essentially moving a few more into keystone-operator). Do we really think other operators will need to share the Keystone specific reason/status codes outside of keystone-operator?

[the remark this answers is not in the document]
Comment (dprince) [reply api/v1beta1/keystoneapi_types.go]: @stuggi Added the status/conditions to lib-common and rebased this patch on the new changes. Also had to update the controller to use the new Secret helpers in lib-common

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**